broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -276,6 +279,35 @@ private void transitionToLeader(final CompletableActorFuture<Void> transitionCom
   }
 
   private ActorFuture<Void> basePartitionInstallation() {
+    final ActorFuture installFuture = new CompletableActorFuture();
+    openLogStream()
+        .onComplete(
+            (log, error) -> {
+              if (error == null) {
+                this.logStream = log;
+                if (!deferredCommitNotifications.isEmpty()) {
+                  deferredCommitNotifications.forEach(Runnable::run);
```

Comment:
Why we need this ? :eyes:

Reply:
I had a deeper thinking about that. I think it should be enough to just keep track of the last commit position, which we received. It is not necessary to keep track of the lambdas, which in the end set the commit position.

Reply:
:+1:  That would be enough.